extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -234,16 +237,40 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		if (!uri) {
 			let openDocument = azdata.nb.activeNotebookEditor;
 			if (openDocument) {
-				bookItem = this.currentBook?.getNotebook(openDocument.document.uri.fsPath);
+				let notebookPath = openDocument.document.uri.fsPath;
+				bookItem = this.currentBook?.getNotebook(notebookPath);
+				// if the node is not expanded getNotebook returns undefined, try getChildren of the book to get the item.
+				if (!bookItem) {
+					// get the parent node and expand it if it's not already
+					let allNodes = this.currentBook?.getAllNotebooks();
+					let book = allNodes ? Array.from(allNodes?.keys())?.filter(x => x.indexOf(notebookPath.substring(0, notebookPath.lastIndexOf(path.sep))) > -1) : undefined;
+					let bookNode = book?.length > 0 ? this.currentBook?.getNotebook(book.find(x => x.substring(0, x.lastIndexOf(path.sep)) === notebookPath.substring(0, notebookPath.lastIndexOf(path.sep)))) : undefined;
+					if (bookNode) {
+						await this._bookViewer.reveal(bookNode, { select: true, focus: false, expand: 3 });
+						bookItem = this.currentBook?.getNotebook(notebookPath);
+					}
+				}
 			}
 		} else if (uri.fsPath) {
+			let notebookPath = uri.fsPath;
 			bookItem = this.currentBook?.getNotebook(uri.fsPath);
+			// if the node is not expanded getNotebook returns undefined, try getChildren of the book to get the item.
+			if (!bookItem) {
```

Comment:
Looks like these code sections are the same besides the initial notebookPath. You could probably move this into a separate block below, and only fill the target notebook path in these if cases.